lib/AST/ASTPrinter.cpp

```diff
@@ -907,14 +907,11 @@ class PrintAST : public ASTVisitor<PrintAST> {
         DC = DC->getParent()->getInnermostTypeContext();
```

Comment:
Or even better, use `Current->getInnermostDeclContext()` for the `mapTypeOutOfContext()` call, and `Current->getDeclContext()` for `getMemberSubstitutions()`. This way it is totally clear what is going on.